llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -8989,6 +8989,15 @@ def : Pat<(shl (i64 (zext GPR32:$Rn)), (i64 imm0_63:$imm)),
                    (i64 (i64shift_a        imm0_63:$imm)),
                    (i64 (i64shift_sext_i32 imm0_63:$imm)))>;
 
+def : Pat<(shl (i64 (and (i64 (anyext GPR32:$Rn)), 0xff)), (i64 imm0_63:$imm)),
```

Comment:
I'm actually quite surprised that the anyext survived all the way through to isel! I was expecting a DAG combine to simply fold `and (i64 (anyext GPR32:$Rn)), 0xff` into something like `and GPR64:$Rn, 0xff`.

Reply:
It's a shame we haven't done that because it would make this pattern simpler. Is this worth looking into?

Reply:
I think it needs to get the types correct.